### PR TITLE
Decorator

### PR DESCRIPTION
**Pull request by danielamoreno699**
In this milestone, I:

- Create a  new class called Nameable used as a Interface. There I implemented a method called correct_name that will raise a NotImplementedError.

- Turned  Person class to Nameable. There the Person class inherits from Nameable and included a correct_name method that return name attribute 

- Inlcuded a base Decorator in a separate file that inherits from  Nameable. There, the constructor was assigned to a neamble object from params to instance variable. Implemented the correct_name method that returns the result of the correct_name method of the @nameable.

- Created a CapitalizeDecorator class that inherits from the base Decorator class. Implemented a method correct_name that capitalizes the output of @nameable.correct_name.

- Create a TrimmerDecorato class that inherits from the base Decorator class. Implemented a method correct_name that makes sure that the characters are less than 10.

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require './nameable'
+
+class BaseDecorator < Nameable
+  def correct_name?
+    @nameable.correct_name?
```
Comment (uwadonat) [inline base_decorator.rb:10]: - [ ] As requested in the project requirement, let's make sure this class has a constructor with a namable object from params and is assigned to an instance variable.

hint: 
def initialize(nameable)
 @nameable = nameable
end

**Review by uwadonat — CHANGES_REQUESTED** [review]
## Hi @danielamoreno699 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Your commit messages are very descriptive ✔️
- Correct workflow was used ✔️
- No linter errors ✔️
- Well-organized and clean project ✔️
almost

## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by danielamoreno699** [comment]
hi @uwadonat , yes. I am aware of that. I added the constructor but the rubocop removes it automaticallyy. that's why i want to ask if i can disable a eslint rule for that special case.

**Comment by danielamoreno699** [comment]
Captura de pantalla 2023-06-20 140028

File: base_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super
+    @nameable = nameable
+  end
```
Comment (KanzaTahreem) [inline base_decorator.rb]: - [ ] Well done for adding the constructor 🥇 However, when I am testing this, the following error is occurring⚠👇 Kindly fix this ✨
 
  image

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @danielamoreno699 🙋‍♀️

Good job so far! I really appreciate the effort you put in!  **_However, there are few this that still need to be fixed. But you are almost there!_** 💪 🏁

so close


## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ Implemented decorator
- ✅ Descriptive PR 
- ✅ Correct Linters

## Required Changes 🛑

_Please check the comments under the review._

## Optional suggestions ❕


N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by laiifuu — APPROVED** [review]
# STATUS: APPROVED :heavy_check_mark: 

## Hi @danielamoreno699 :wave: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

friends-joey


## Highlights
- [x] Required changes implemented :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**